### PR TITLE
CBL-4600: Doc update c4repl_start

### PR DESCRIPTION
**Pull request by jianminzhao**
Added the following
```
\note This function may perform action in the calling thread that requires the transaction from the database
              that was passed to the replicator instance, \p repl.
```
I merely note the fact that is critical to the user I assume that general prerequisite knowledge in using the library that they cannot open a transaction inside a transaction.

### CODE REVIEW

**Review by borrrden — CHANGES_REQUESTED** [review]
This is kind of a word salad.  It's hard to follow even if you know the situation, and I think it will be impossible to follow if you don't.  Try to be clear about what to avoid.

**Comment by velicuvlad** [comment]
Maybe something like `If this function is called while a transaction is open on the same database it will deadlock, as it trying to acquire the same database` ... maybe worth specifying that this is considered transaction as well (starting the repl)...?

**Comment by jianminzhao** [comment]
Simplified wording. Thanks to @velicuvlad for the suggestion. I avoided emphasizing the "same thread," because it makes the doc unnecessary complicated and why people want to do transaction in another thread while calling this function. If you think we should limit the "Do not" to the "same thread," I can change it.

**Review by borrrden — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Review by callumbirks — APPROVED** [review]

**Review by velicuvlad — APPROVED** [review]
